test/test_mkldnn.py

```diff
@@ -299,27 +298,34 @@ def _test_conv_bf16_base(self, dim):
                                     dilation=dilation,
                                     bias=bias,
                                     groups=groups).float()
-            x_bf16 = x.bfloat16()
-            if torch.ops.mkldnn._is_mkldnn_bf16_supported():
+            x_lower = x.to(dtype=dtype)
+            if (dtype == torch.bfloat16 and torch.ops.mkldnn._is_mkldnn_bf16_supported()) or \
+               (dtype == torch.half and torch.ops.mkldnn._is_mkldnn_fp16_supported()):
                 mkldnn_conv = mkldnn_utils.to_mkldnn(copy.deepcopy(conv))
-                mkldnn_conv_bf16 = mkldnn_utils.to_mkldnn(copy.deepcopy(conv), torch.bfloat16)
+                mkldnn_conv_lower = mkldnn_utils.to_mkldnn(copy.deepcopy(conv), dtype)
                 y = mkldnn_conv(x.to_mkldnn()).to_dense()
-                y_bf16 = mkldnn_conv_bf16(x_bf16.to_mkldnn()).to_dense(torch.float32)
-                self.assertEqual(y, y_bf16, atol=1e-1, rtol=1e-3)
+                y_lower = mkldnn_conv_lower(x_lower.to_mkldnn()).to_dense(torch.float32)
+                self.assertEqual(y, y_lower, atol=1e-1, rtol=1e-3)
             else:
-                msg = r"bf16 path needs the cpu support avx512bw, avx512vl and avx512dq"
-                with self.assertRaisesRegex(RuntimeError, msg):
-                    mkldnn_conv_bf16 = mkldnn_utils.to_mkldnn(copy.deepcopy(conv), torch.bfloat16)
-                    y_bf16 = mkldnn_conv_bf16(x_bf16.to_mkldnn()).to_dense(torch.float32)
-
-    def test_conv1d_bf16(self):
-        self._test_conv_bf16_base(dim=1)
-
-    def test_conv2d_bf16(self):
-        self._test_conv_bf16_base(dim=2)
-
-    def test_conv3d_bf16(self):
-        self._test_conv_bf16_base(dim=3)
+                msg = {
+                    torch.bfloat16: r"bf16 path needs the cpu support avx_ne_convert or avx512bw, avx512vl and avx512dq",
+                    torch.half: r"fp16 path needs the cpu support avx_ne_convert or avx512_fp16",
+                }
+                with self.assertRaisesRegex(RuntimeError, msg[dtype]):
+                    mkldnn_conv_lower = mkldnn_utils.to_mkldnn(copy.deepcopy(conv), dtype)
+                    y_lower = mkldnn_conv_lower(x_lower.to_mkldnn()).to_dense(torch.float32)
+
+    def test_conv1d_lower_precision(self):
+        self._test_conv_lower_precision_base(dim=1, dtype=torch.bfloat16)
+        self._test_conv_lower_precision_base(dim=1, dtype=torch.half)
+
+    def test_conv2d_lower_precision(self):
+        self._test_conv_lower_precision_base(dim=2, dtype=torch.bfloat16)
+        self._test_conv_lower_precision_base(dim=2, dtype=torch.half)
+
+    def test_conv3d_lower_precision(self):
```

Comment:
nit: If you make these dtypes arguments to the tests and use a dtype decorator it'll be easier to see if one of them fails. Otherwise you'll get one failure for either one of the dtypes.

Reply:
Thanks for the comments. I will improve such test cases in https://github.com/pytorch/pytorch/pull/99497 since it also involves these tests.